### PR TITLE
S05 us153 kingdoms

### PR DESCRIPTION
**Pull request by boubseb**

### CODE REVIEW

**Review by OCR-VH-DevWeb — APPROVED** [review]
Approved

File: src/main/java/com/example/demo/entity/City.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @JoinColumn(name = "city_id")
+    private Region region;
+
+    public City(){}
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
```
Comment (Benja-san) [inline src/main/java/com/example/demo/entity/City.java:29]: remove setter for id

File: src/main/java/com/example/demo/entity/Belief.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    @ManyToOne
+    @JoinColumn(name = "belief_id")
+    private Kingdom kingdom;
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
```
Comment (Benja-san) [inline src/main/java/com/example/demo/entity/Belief.java:25]: remove setter for id

File: src/main/java/com/example/demo/entity/Divinity.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        this.kingdom = kingdom;
+    }
+
+    public Divinity() {}
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
```
Comment (Benja-san) [inline src/main/java/com/example/demo/entity/Divinity.java:40]: remove setter for id

File: src/main/java/com/example/demo/entity/Divinity.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+public class Divinity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String name;
+    private String type;
+    private String gender;
+    private String statistics;
+    private String effects;
```
Comment (Benja-san) [inline src/main/java/com/example/demo/entity/Divinity.java:20]: if it is a string, call it singular

File: src/main/java/com/example/demo/entity/Divinity.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+@Entity
+public class Divinity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String name;
+    private String type;
+    private String gender;
+    private String statistics;
```
Comment (Benja-san) [inline src/main/java/com/example/demo/entity/Divinity.java:19]: if it is a string, call it singular

File: src/main/java/com/example/demo/entity/Divinity.java (hunk 4 above)
Comment (Benja-san) [reply src/main/java/com/example/demo/entity/Divinity.java:20]: If it is a list, use List type

File: src/main/java/com/example/demo/entity/Kingdom.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    @OneToMany(mappedBy = "kingdom")
+    List<Divinity> divinities = new ArrayList<Divinity>();
+
+    @OneToMany(mappedBy = "kingdom")
+    List<Belief> believes = new ArrayList<Belief>();
+    
+    public Long getId() {
+        return id;
+    }
+    public void setId(Long id) {
```
Comment (Benja-san) [inline src/main/java/com/example/demo/entity/Kingdom.java:35]: remove setter for id

File: src/main/java/com/example/demo/entity/Region.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+    public void setKingdom(Kingdom kingdom) {
+        this.kingdom = kingdom;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
```
Comment (Benja-san) [inline src/main/java/com/example/demo/entity/Region.java:43]: remove setter for id

File: src/main/java/com/example/demo/repository/BeliefRepository.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package com.example.demo.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import com.example.demo.entity.Belief;
+
+public interface BeliefRepository extends JpaRepository<Belief, Long> {
```
Comment (Benja-san) [inline src/main/java/com/example/demo/repository/BeliefRepository.java:7]: usez repository annotation

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Pull request closed without merging**